tests/unit/src/test/scala/tests/BspBuildChangedSuite.scala

```diff
@@ -0,0 +1,74 @@
+package tests
+
+import scala.meta.internal.metals.ServerCommands
+
+import scala.util.Failure
+import scala.util.Success
+
+class BspBuildChangedSuite
+    extends BaseLspSuite("bsp-build-changed", ScalaCliBuildInitializer) {
```

Comment:
Out of curiosity, why are we using `ScalaCliBuildInitialiazer` for this? Are we indirectly starting to add tests for Scala CLI as a way to test this change, or would this also just be testable with another existing BSP server? If the latter is true, then i'd rather do that and leave this addition to another PR that is about Scala CLI.

Reply:
I needed a BSP server emitting `buildTarget/didChange` notifications. AFAIK, Bloop doesn't emit those. And quickbuild doesn't either it seems. (Not sure about sbt, but it's more heavyweight.)

The test downloads a `0.0.1` launcher, so not a nightly. And as soon as `0.0.2` is cut, we can download the class path of Scala CLI from Maven Central, so that it's even less likely to break in the future.

Reply:
If there's a reasonably straightforward way of adding `buildTarget/didChange` support to quick build, I'd rather do that actually.

Reply:
> I needed a BSP server emitting buildTarget/didChange notifications. AFAIK, Bloop doesn't emit those. And quickbuild doesn't either it seems. (Not sure about sbt, but it's more heavyweight.)

No Bloop for sure doesn't. Actually thinking about this more I thought for sure sbt would, but I see during the `build/initialize` with sbt, it doesn't declare it under `capabilities`.

> If there's a reasonably straightforward way of adding buildTarget/didChange support to quickbuild, I'd rather do that actually.

Does `buildTarget/didChange` make sense actually when the build server writes to disk? I think the main reason that Bloop doesn't support it is because of that, since there is no concept of a reload or anything to trigger after a `didChange`, but rather it does a fresh export each time. Looking at `bill`, it might be somewhat straight forward to try to implement this there.

Reply:
We use Bill to test out most of the build server specific stuff.

Reply:
Bill doesn't support dependencies AFAIK… That means, dependency support has to be added to Bill just to test that feature. I'm not sure it's worth the hassle…